### PR TITLE
Allow for passing in function to generate array items

### PR DESCRIPTION
**Pull request by Maximilianos**
Slightly better way of doing something like:

``` javascript
filledArray(index => index * 10, 5) // = [0, 10, 20, 30, 40, 50]
// vs.
filledArray(0, 5).map((_, index) => index * 10) // = [0, 10, 20, 30, 40, 50]
```

or

``` javascript
filledArray(() => Math.random(), 10000);
```

### CODE REVIEW

**Comment by wbinnssmith** [comment]
This was my first thought when stumbling on this module. IMHO this makes a ton of sense without unnecessarily expanding the scope of this module. :+1:

File: index.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
 'use strict';
-module.exports = function (str, n) {
+module.exports = function (item, n) {
 	var ret = new Array(n);
+	var isCb = typeof item === 'function';
```
Comment (kevva) [inline index.js]: `isFn` is a better name imo.

**Comment by kevva** [comment]
:+1:

File: test.js [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+	return index + 1;
+}
+
+function fizzBuzz(index) {
+	// https://gist.github.com/jaysonrowe/1592432#gistcomment-882551
+	return (++index % 3 ? '' : 'Fizz') + (index % 5 ? '' : 'Buzz') || index;
+}
+
+function comprehensive(index, length, partial) {
+	return partial.indexOf(index) >= 0 ? length : index + 1;
```
Comment (kevva) [inline test.js]: `partial.includes(index)`

**Comment by kevva** [comment]
:+1:

**Pull request closed without merging**